src/main/java/io/zeebe/kafka/connect/sink/ZeebeSinkTask.java

```diff
@@ -82,7 +83,8 @@ public String version() {
         sinkRecords
             .stream()
             .map(r -> this.preparePublishRequest(client, r))
-            .map(FinalCommandStep::send)
+            .map(ZeebeSinkFuture::new)
+            .map(ZeebeSinkFuture::executeAsync)
             .toArray(CompletableFuture[]::new);
 
     return CompletableFuture.allOf(inFlightRequests);
```

Comment:
Did you check if we need to block here in the end?

Reply:
In my test all processes where completed even with back pressure kicking in. I think as every future now has an endless retry it's fine not to block.